aten/src/ATen/native/cuda/UnarySignKernels.cu

```diff
@@ -52,21 +53,45 @@ void signbit_kernel_cuda(TensorIteratorBase& iter){
 }
 
 template<typename T>
-__host__ __device__ static inline c10::complex<T> sgn_wrapper(c10::complex<T> z) {
+C10_HOST_DEVICE static inline c10::complex<T> sgn_wrapper(c10::complex<T> z) {
   if (z == c10::complex<T>(0, 0)) {
     return c10::complex<T>(0, 0);
   } else {
     return z / std::abs(z);
   }
 }
 
+const char sgn_name[] = "sgn_kernel";
 void sgn_kernel_cuda(TensorIteratorBase& iter){
-  AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "sgn_cuda", [&]() {
+  auto dtype = iter.dtype();
+  #if AT_USE_JITERATOR()
+    static const auto sgn_string = jiterator_stringify(
+        template <typename T>
+        T sgn_kernel(T a) {
+          const T z = T(0);
```

Comment:
It would be nice if you rename `z` to `zero` (`z` is many times used to refer a complex number). Also it would be nice if the names are consistent with wrapper above so that it is easy verify that both are same.

Thanks!

Reply:
Agreed with you :) I've updated it. Thanks!!